src/AdoNetCore.AseClient/AseDataReader.cs

```diff
@@ -143,37 +145,45 @@ public object GetValue(int i)
             throw new ArgumentOutOfRangeException();
         }
 
-        public int GetValues(object[] values)
+        public override int GetValues(object[] values)
         {
             throw new NotImplementedException();
         }
 
-        public bool IsDBNull(int i)
+        public override bool IsDBNull(int i)
         {
             throw new NotImplementedException();
         }
 
-        public int FieldCount => _currentResult >= 0 && _currentResult < _results.Length
+        public override int FieldCount => _currentResult >= 0 && _currentResult < _results.Length
             ? _results[_currentResult].Formats.Length
             : 0;
 
-        object IDataRecord.this[int i] => GetValue(i);
+        public override bool HasRows => _results.Length > 0;
 
-        object IDataRecord.this[string name] => GetValue(GetOrdinal(name));
-        
-        public void Dispose() { }
+        public override object this[int ordinal] => GetValue(ordinal);
 
-        public void Close()
+        public override object this[string name] => GetValue(GetOrdinal(name));
+
+#if NETCOREAPP2_0 || NET45
+        public override void Close() { }
+#else
+        public void Close() { }
+#endif
+
+#if NETCOREAPP2_0 || NET45
```

Comment:
Might be better to explicitly state the old versions, which won't change, and then the else will include all future versions. Stating the the future versions explicitly means you'll have to fix this list each time a new version comes out.

Reply:
Yeah, we might be able to do it by adding something like this to the .csproj:
````
<PropertyGroup Condition="'$(TargetFramework)' == 'NET451' Or '$(TargetFramework)' == 'NET452' Or '$(TargetFramework)' == 'NET46' Or '$(TargetFramework)' == 'NET461' Or '$(TargetFramework)' == 'NET462'">
    <DefineConstants>$(DefineConstants);NET45</DefineConstants>
</PropertyGroup>
````

I'll raise a separate issue to deal with it